File: Chap5-ModelViewArchitectureAdvanced/5-8CustomModelsEditableWithHeadersDemo/person.cpp
#include "person.h"
#include <QDebug>


Person::Person(QObject *parent) : QObject(parent)
{

}

Person::Person(const QString &names, const QString &favoritecolor, const int &age, QObject *parent):
    QObject(parent),m_names(names),m_favoriteColor(favoritecolor),m_age(age)
{

}

QString Person::names() const
{
    return m_names;
}

QString Person::favoriteColor() const
{
    return m_favoriteColor;
}

int Person::age() const
{
    return m_age;
}

void Person::setNames(QString names)
{
    if (m_names == names)
        return;

    m_names = names;
    emit namesChanged(m_names);
}

void Person::setFavoriteColor(QString favoriteColor)
{
    qDebug() << "Favorite color called";
    if (m_favoriteColor == favoriteColor)
        return;

    m_favoriteColor = favoriteColor;
    emit favoriteColorChanged(m_favoriteColor);
}

void Person::setAge(int age)
{
    if (m_age == age)
        return;

    m_age = age;
    emit ageChanged(m_age);
}

File: Chap5-ModelViewArchitectureAdvanced/5-10CustomModelsCustomRolesDemo/person.py
from PySide6.QtCore import QObject, Signal

class Person(QObject):
    namesChanged = Signal(str)
    favoriteColorChanged = Signal(str)
    ageChanged = Signal(int)
    
    def __init__(self, names="", favorite_color="", age=0, parent=None):
        super().__init__(parent)
        self._names = names
        self._favorite_color = favorite_color
        self._age = age
    
    def names(self):
        """Get the person's name"""
        return self._names
    
    def favoriteColor(self):
        """Get the person's favorite color"""
        return self._favorite_color
    
    def age(self):
        """Get the person's age"""
        return self._age
    
    def setNames(self, names):
        """Set the person's name"""
        if self._names == names:
            return
        
        self._names = names
        self.namesChanged.emit(self._names)
    
    def setFavoriteColor(self, favorite_color):
        """Set the person's favorite color"""
        print("Favorite color called")
        if self._favorite_color == favorite_color:
            return
        
        self._favorite_color = favorite_color
        self.favoriteColorChanged.emit(self._favorite_color)
    
    def setAge(self, age):
        """Set the person's age"""
        if self._age == age:
            return
        
        self._age = age
        self.ageChanged.emit(self._age)